### ISSUE TITLE
Using web project with SQLite db without VS tooling

### ISSUE THREAD

**Issue by peterblazejewicz** [OmniSharp/generator-aspnet#713]
How the users are supposed to start using web project - which uses SQLite - where there is no built-in automatic migration now?

The project builds, runs, but at the very first time where EF is to be used as backing store, the runtime exception will occur.

TBI

@rustd
@sayedihashimi

**Comment by peterblazejewicz** [comment]
/cc
@natemcmaster

**Comment by rustd** [comment]
What is the runtime exception? if you see an error page that tells you to apply migrations and refresh the page then it is expected. This is the same behavior as VS templates

**Comment by rustd** [comment]
The generators should set the env to be development. This is what the VS templates do. Maybe ask the user at the time of project creation or output a message letting the user know that they should set the env to be development for dev only scenario.

**Comment by peterblazejewicz** [comment]
The dotnet needs a simple solution for this.
/cc
@sayedihashimi 

TL;DR;
@rowanmiller @glennc 
It seems that using `dotnet ef database update` before `dotnet run` is enough to avoid that all mess. May I suppose that we add this to `generator-aspnet` hints and into README.md as distinct diff from RC1? This will allow to still use default `Production` settings


1) 'Production' is default, documented, can't be set because of implementation detail (consistent experience x-plat) I think
2) I think the generator should install NPM and Bower deps now, if we expect that users will have to go into development mode at some point when running web templates. Most of the generators for web apps always do that on the end of project scaffolding

Notes:
- by default no migration is applied to backing EF storage (compared to RC1)
- everyone (except ones following changes in RC2 and Templates) will hit that problem when using Register/Login actions for the first time
- by default 'Production' is enabled so after using Login/Register actions users see:
```
Error.
An error occurred while processing your request.
Development Mode
Swapping to Development environment will display more detailed information about the error that occurred.
Development environment should not be enabled in deployed applications, as it can result in sensitive information from exceptions being displayed to end users. For local debugging, development environment can be enabled by setting the ASPNETCORE_ENVIRONMENT environment variable to Development, and restarting the application.
```
NOTE> Even that information could be probably missing in next iteration. See: https://github.com/aspnet/Templates/issues/560

- so you have to switch to Development mode (unless you're smart enough to look into console output:
```
info: Microsoft.EntityFrameworkCore.Storage.Internal.RelationalCommandBuilderFactory[1]
      Executed DbCommand (462ms) [Parameters=[@__normalizedUserName_0='?'], CommandType='Text', CommandTimeout='30']
      SELECT "u"."Id", "u"."AccessFailedCount", "u"."ConcurrencyStamp", "u"."Email", "u"."EmailConfirmed", "u"."LockoutEnabled", "u"."LockoutEnd", "u"."NormalizedEmail", "u"."NormalizedUserName", "u"."PasswordHash", "u"."PhoneNumber", "u"."PhoneNumberConfirmed", "u"."SecurityStamp", "u"."TwoFactorEnabled", "u"."UserName"
      FROM "AspNetUsers" AS "u"
      WHERE "u"."NormalizedUserName" = @__normalizedUserName_0
      LIMIT 1
fail: Microsoft.EntityFrameworkCore.Query.RelationalQueryCompilationContextFactory[1]
      An exception occurred in the database while iterating the results of a query.
      Microsoft.Data.Sqlite.SqliteException: SQLite Error 1: 'no such table: AspNetUsers'.
```
but again, the templates are for noobs by design
- to switch into development mode [you're lucky enough to know to execute](http://stackoverflow.com/questions/35967682/asp-net-core-ignores-aspnet-env-and-hostingenvironment/35967920#35967920):
```
export ASPNETCORE_ENVIRONMENT=Development
```
- once restarted **user have to explicitely cause the same error again** to see solution (`to see the solution break things again`):
```
A database operation failed while processing the request.

AggregateException: One or more errors occurred. (SQLite Error 1: 'no such table: AspNetUsers'.) 
SqliteException: SQLite Error 1: 'no such table: AspNetUsers'. 
Applying existing migrations for ApplicationDbContext may resolve this issue
There are migrations for ApplicationDbContext that have not been applied to the database

00000000000000_CreateIdentitySchema
Apply Migrations

In Visual Studio, you can use the Package Manager Console to apply pending migrations to the database:

PM> Update-Database
Alternatively, you can apply pending migrations from a command prompt at your project directory:

> dotnet ef database update
```

**Comment by sayedihashimi** [comment]
>May I propose that we add this to  generator-aspnet  hints and into README.md as distinct diff from RC1?

Sounds great.

>I think the generator should install NPM and Bower deps now, if we expect that users will have to go into development mode at some point when running web templates. Most of the generators for web apps always do that on the end of project scaffolding

If that's how most other generators work then lets do it.

I'm not sure about the `ASPNETCORE_ENVIRONMENT` env var issue. In VS we automatically set it to `debug`, but we can't really do that for command line users in the same way. Is this a problem for other frameworks and if so how do they solve it? When running `yo aspnet` for the first time, would it make sense to ask if users want us to update their profile to set the env var?

**Comment by peterblazejewicz** [comment]
@sayedihashimi 
OK
- add hint for web (using SQLite) to run `dotnet ef database udpate`
- add small NOTE into README why this is required
- update postretore scripts to run: `npm install && bower install` as `postrestore` script
I'll update RC2 with above ones

Thanks!

**Comment by natemcmaster** [comment]
Not sure what is in the templates, but this may be relevant:

When using SQLite with EF, relative connection strings are relative to the application base directory. e.g. "Filename=./blog.db". This directory changes with the upgrade from DNX to .NET Core CLI.

```
MyProject/  <-- the app base directory in DNX RC1
    project.json
    bin/
         Debug/
             netcoreapp1.0/   <--- the app base directory in .NET Core CLI Preview1
                   MyProject.dll 
```
If you are using .NET Core CLI and want to use a pre-existing sqlite db file, make sure your build options put the db file in the output path (project.json > "buildOptions" > "copyToOutput")

**Comment by peterblazejewicz** [comment]
@natemcmaster 
> Not sure what is in the templates [...]

https://github.com/OmniSharp/generator-aspnet/blob/rc2/templates/projects/web/appsettings.json#L3
```
"ConnectionStrings": {
    "DefaultConnection": "Data Source=<%= namespace %>.db"
  },
```
and this is read by configuration builder (usual stuff).

**Comment by peterblazejewicz** [comment]
Closed via:
203dd97 and ecb2b27
Thanks!
